silo-vaults/contracts/MetaMorpho.sol

```diff
@@ -480,6 +482,20 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
         emit EventsLib.Skim(_msgSender(), token, amount);
     }
 
+    function claimRewards() public {
+        (address[] memory logics) = INCENTIVES_MODULE.getAllIncentivesClaimingLogics();
+
+        require(!_lock, ErrorsLib.ClaimingRewardsError());
+        _lock = true;
```

Comment:
We can create a modifier for the 'lock' functionality to avoid code duplication. Something like this:
```
modifier Lock() {
        require(!_lock, ErrorsLib.NotificationDispatchError());
        _lock = true;
        
        _;
        
        _lock = false;
}
```
And use it for `claimRewards` and `_afterTokenTransfer` functions